data_structures/rb_tree2.cpp

```diff
@@ -0,0 +1,1490 @@
+/**
+ * @file rbtree2.cpp
+ * @author Brennan Allen (brennandoubt@gmail.com)
```

Comment:
This is to add your GitHub profile as well. :)
```suggestion
 * @author [Brennan Allen](https://github.com/brennandoubt) (brennandoubt@gmail.com)
```